subsys/mgmt/osdp/src/osdp_pd.c

```diff
@@ -189,6 +189,69 @@ static bool do_command_callback(struct osdp_pd *pd, struct osdp_cmd *cmd)
 	return true;
 }
 
+static int pd_cmd_cap_ok(struct osdp_pd *pd, struct osdp_cmd *cmd)
+{
+	struct osdp_pd_cap *cap = NULL;
+
+	/* Validate the cmd_id against a PD capabilities where applicable */
+	switch (pd->cmd_id) {
+	case CMD_ISTAT:
+		cap = &pd->cap[OSDP_PD_CAP_CONTACT_STATUS_MONITORING];
+		if (cap->num_items == 0 || cap->compliance_level == 0) {
+			break;
+		}
+		return 0; /* Remove this when REPLY_ISTATR is supported */
+	case CMD_OSTAT:
+		cap = &pd->cap[OSDP_PD_CAP_OUTPUT_CONTROL];
+		if (cap->num_items == 0 || cap->compliance_level == 0) {
+			break;
+		}
+		return 0; /* Remove this when REPLY_OSTATR is supported */
+	case CMD_OUT:
+		cap = &pd->cap[OSDP_PD_CAP_OUTPUT_CONTROL];
+		if (!cmd || cap->compliance_level == 0 ||
+		    cmd->output.output_no + 1 > cap->num_items) {
+			break;
+		}
+		return 1;
+	case CMD_LED:
+		cap = &pd->cap[OSDP_PD_CAP_READER_LED_CONTROL];
+		if (!cmd || cap->compliance_level == 0 ||
+		    cmd->led.led_number + 1 > cap->num_items) {
+			break;
+		}
+		return 1;
+	case CMD_BUZ:
+		cap = &pd->cap[OSDP_PD_CAP_READER_AUDIBLE_OUTPUT];
+		if (cap->num_items == 0 || cap->compliance_level == 0) {
+			break;
+		}
+		return 1;
+	case CMD_TEXT:
+		cap = &pd->cap[OSDP_PD_CAP_READER_TEXT_OUTPUT];
+		if (cap->num_items == 0 || cap->compliance_level == 0) {
+			break;
+		}
+		return 1;
+	case CMD_CHLNG:
+	case CMD_SCRYPT:
+	case CMD_KEYSET:
+		cap = &pd->cap[OSDP_PD_CAP_COMMUNICATION_SECURITY];
+		if (cap->compliance_level == 0) {
+			pd->reply_id = REPLY_NAK;
+			pd->ephemeral_data[0] = OSDP_PD_NAK_SC_UNSUP;
+			return 0;
+		}
+		return 1;
+	}
+
+	pd->reply_id = REPLY_NAK;
+	pd->ephemeral_data[0] = OSDP_PD_NAK_CMD_UNKNOWN;
+	LOG_INF("PD is not capable of handling CMD(%02x); "
```

Comment:
LOG_INF feels wrong here, maybe error/warn/dbg?

Reply:
Yes, changing this to an error level log.